#!/usr/bin/env python
PACKAGE = "phidgets"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

logging = gen.enum([gen.const("None",     int_t, 0, "No logging"),
                    gen.const("Critical", int_t, 1, "Critical logging"),
                    gen.const("Error",    int_t, 2, "Error logging"),
                    gen.const("Warning",  int_t, 3, "Warning logging"),
                    gen.const("Info",     int_t, 4, "Info logging"),
                    gen.const("Debug",    int_t, 5, "Debug logging"),
                    gen.const("Verbose",  int_t, 6, "Verbose logging")],
                    "Phidgets logging level")

gen.add("logging", int_t, 0, "Level of Phidgets logging", 4, 0, 6, edit_method=logging)
gen.add("console", bool_t, 1, "Log to console (otherwise to file)", True)
gen.add("rotate",  bool_t, 2, "Rotate the log file", True)
gen.add("rotating_size", int_t, 2, "The file size above which the file should be rotated in bytes", 10485760, 32768, 10485760)
gen.add("rotating_max_files", int_t, 2, "The number of log files that should be kept after rotation", 1, 0, 64)

exit(gen.generate(PACKAGE, "phidgets", "Logging"))